src/libraries/System.Text.Json/tests/Serialization/SerializationWrapper.cs

```diff
@@ -19,96 +19,90 @@ public abstract class SerializationWrapper
+        protected internal abstract Task<string> SerializeAsync<T>(T value, JsonSerializerOptions options = null);
 
 
         private class SpanSerializerWrapper : SerializationWrapper
         {
-            protected internal override string Serialize(object value, Type inputType, JsonSerializerOptions options = null)
+            protected internal override Task<string> SerializeAsync(object value, Type inputType, JsonSerializerOptions options = null)
             {
                 byte[] result = JsonSerializer.SerializeToUtf8Bytes(value, inputType, options);
-                return Encoding.UTF8.GetString(result);
+                return Task.FromResult(Encoding.UTF8.GetString(result));
             }
 
-            protected internal override string Serialize<T>(T value, JsonSerializerOptions options = null)
+            protected internal override Task<string> SerializeAsync<T>(T value, JsonSerializerOptions options = null)
             {
                 byte[] result = JsonSerializer.SerializeToUtf8Bytes<T>(value, options);
-                return Encoding.UTF8.GetString(result);
+                return Task.FromResult(Encoding.UTF8.GetString(result));
             }
         }
 
         private class StringSerializerWrapper : SerializationWrapper
         {
-            protected internal override string Serialize(object value, Type inputType, JsonSerializerOptions options = null)
+            protected internal override Task<string> SerializeAsync(object value, Type inputType, JsonSerializerOptions options = null)
             {
-                return JsonSerializer.Serialize(value, inputType, options);
+                return Task.FromResult(JsonSerializer.Serialize(value, inputType, options));
             }
 
-            protected internal override string Serialize<T>(T value, JsonSerializerOptions options = null)
+            protected internal override Task<string> SerializeAsync<T>(T value, JsonSerializerOptions options = null)
             {
-                return JsonSerializer.Serialize(value, options);
+                return Task.FromResult(JsonSerializer.Serialize(value, options));
             }
         }
 
         private class StreamSerializerWrapper : SerializationWrapper
         {
-            protected internal override string Serialize(object value, Type inputType, JsonSerializerOptions options = null)
+            protected internal override async Task<string> SerializeAsync(object value, Type inputType, JsonSerializerOptions options = null)
             {
-                return Task.Run(async () =>
-                {
-                    using var stream = new MemoryStream();
-                    await JsonSerializer.SerializeAsync(stream, value, inputType, options);
-                    return Encoding.UTF8.GetString(stream.ToArray());
-                }).GetAwaiter().GetResult();
+                using var stream = new MemoryStream();
```

Comment:
Again, this is the key area of code where we switch the wrapper to use await without Task.Run.